jaeger-core/src/main/java/com/uber/jaeger/metrics/InMemoryMetricsFactory.java

```diff
@@ -0,0 +1,175 @@
+  private Map<String, AtomicLong> counters = new ConcurrentHashMap<String, AtomicLong>();
+  private Map<String, AtomicLong> timers = new ConcurrentHashMap<String, AtomicLong>();
+  private Map<String, AtomicLong> gauges = new ConcurrentHashMap<String, AtomicLong>();
+
+  @Override
+  public Counter createCounter(String name, Map<String, String> tags) {
+    final AtomicLong value = new AtomicLong(0);
+    counters.put(Metrics.addTagsToMetricName(name, tags), value);
+
+    return new Counter() {
+      @Override
+      public void inc(long delta) {
+        value.addAndGet(delta);
+      }
+    };
+  }
+
+  @Override
+  public Timer createTimer(final String name, final Map<String, String> tags) {
+    final AtomicLong value = new AtomicLong(0);
+    timers.put(Metrics.addTagsToMetricName(name, tags), value);
+
+    return new Timer() {
+      @Override
+      public void durationMicros(long time) {
+        value.addAndGet(time);
+      }
+    };
+  }
+
+  @Override
+  public Gauge createGauge(final String name, final Map<String, String> tags) {
+    final AtomicLong value = new AtomicLong(0);
+    gauges.put(Metrics.addTagsToMetricName(name, tags), value);
+
+    return new Gauge() {
+      @Override
+      public void update(long amount) {
+        value.addAndGet(amount);
+      }
+    };
+  }
+
+  /**
+   * Returns the counter value information for the counter with the given metric name.
+   * Note that the metric name is not the counter name, as a metric name usually includes the tags.
+   *
+   * @see Metrics#addTagsToMetricName(String, Map)
+   * @param name the metric name, which includes the tags
+   * @param tags the metric tags as comma separated list of entries, like "foo=bar,baz=qux"
+   * @return the counter value or -1, if no counter exists for the given metric name
+   */
+  public long getCounter(String name, String tags) {
+    return getValue(counters, name, tags);
+  }
+
+  /**
+   * Returns the counter value information for the counter with the given metric name.
+   * Note that the metric name is not the counter name, as a metric name usually includes the tags.
+   *
+   * @see Metrics#addTagsToMetricName(String, Map)
+   * @param name the metric name, which includes the tags
+   * @param tags the metric tags
+   * @return the counter value or -1, if no counter exists for the given metric name
+   */
+  public long getCounter(String name, Map<String, String> tags) {
+    return getValue(counters, name, tags);
+  }
+
+  /**
+   * Returns the current value for the gauge with the given metric name. Note that the metric name is not the gauge
+   * name, as a metric name usually includes the tags.
+   *
+   * @see Metrics#addTagsToMetricName(String, Map)
+   * @param name the metric name, which includes the tags
+   * @param tags the metric tags as comma separated list of entries, like "foo=bar,baz=qux"
+   * @return the gauge value or -1, if no gauge exists for the given metric name
```

Comment:
`-1` is a valid gauge value, returning null would be better in this case

Reply:
This class is replacing InMemoryStatsReporter and the `getValue()` is a new method, to help in the tests. The javadoc for the metrics factory clearly says it's not for production purposes, it doesn't make sense to implement a more complex solution for a case that isn't being used at all, and probably never will.

Reply:
As a consumer of this API I would find it more intuitive to check on `null` than in `-1`. What if our test  will use `-1` as a valid number?